crawl-ref/source/mon-place.cc

```diff
@@ -112,6 +112,135 @@ static bool _hab_requires_mon_flight(dungeon_feature_type g)
     return g == DNGN_LAVA || g == DNGN_DEEP_WATER;
 }
 
+enum habital_feature_type_flag
```

Comment:
I think this should be "habitable" not "habital".

I think this is a really good improvement, separately to the summoning changes, as the old system was rather clunky; I am just not sure on a first look at this whether this slightly affects the way e.g. amphibious creatures are placed w.r.t trying to place them in their preferred habitat?

Reply:
Thanks that is a spelling mistake I've fixed It. The code is still using the old habitat code to decide where to place monsters so this shouldn't affect that. Although, we could refactor everywhere is to use this new habitat code if that would be helpful (in a separate PR of cause).